### PR TITLE
Added the zombie_resource podman run in run_policy.py

### PR DESCRIPTION
**Pull request by athiruma**

### CODE REVIEW

File: jenkins/Jenkinsfile [hunk 1]
```diff
@@ -10,6 +10,8 @@ pipeline {
         AWS_ACCESS_KEY_ID_RH_PERF = credentials('cloud-governance-aws-access-key-id-rh-perf')
         AWS_SECRET_ACCESS_KEY_RH_PERF = credentials('cloud-governance-aws-secret-access-key-rh-perf')
         BUCKET_RH_PERF = credentials('cloud-governance-bucket-rh-perf')
+        DELETE_AWS_ACCESS_KEY_ID_PERF = credentials('delete-cloud-governance-aws-access-key-id-perf')
+        DELETE_AWS_SECRET_ACCESS_KEY_PERF = credentials('delete_cloud-governance-aws-secret-access-key-perf')
```
Comment (ebattat) [inline jenkins/Jenkinsfile:14]: delete_cloud-governance-aws-secret-access-key-perf => cloud-governance-aws-secret-access-key-delete-perf
DELETE_AWS_SECRET_ACCESS_KEY_PERF =>  AWS_SECRET_ACCESS_KEY_DELETE_PERF

File: jenkins/Jenkinsfile [hunk 2]
```diff
@@ -10,6 +10,8 @@ pipeline {
         AWS_ACCESS_KEY_ID_RH_PERF = credentials('cloud-governance-aws-access-key-id-rh-perf')
         AWS_SECRET_ACCESS_KEY_RH_PERF = credentials('cloud-governance-aws-secret-access-key-rh-perf')
         BUCKET_RH_PERF = credentials('cloud-governance-bucket-rh-perf')
+        DELETE_AWS_ACCESS_KEY_ID_PERF = credentials('delete-cloud-governance-aws-access-key-id-perf')
```
Comment (ebattat) [inline jenkins/Jenkinsfile:13]: DELETE_AWS_ACCESS_KEY_ID_PERF =>  AWS_ACCESS_KEY_ID_DELETE_PERF
delete-cloud-governance-aws-access-key-id-perf => cloud-governance-aws-access-key-id-delete-perf

File: jenkins/run_policies.py [hunk 3]
```diff
@@ -10,6 +10,8 @@
 AWS_ACCESS_KEY_ID_RH_PERF = os.environ['AWS_ACCESS_KEY_ID_RH_PERF']
 AWS_SECRET_ACCESS_KEY_RH_PERF = os.environ['AWS_SECRET_ACCESS_KEY_RH_PERF']
 BUCKET_RH_PERF = os.environ['BUCKET_RH_PERF']
+DELETE_AWS_ACCESS_KEY_ID_PERF = os.environ['DELETE_AWS_ACCESS_KEY_ID_PERF']
```
Comment (ebattat) [inline jenkins/run_policies.py:13]: DELETE_AWS_ACCESS_KEY_ID_PERF => AWS_ACCESS_KEY_ID_DELETE_PERF

File: jenkins/run_policies.py [hunk 4]
```diff
@@ -10,6 +10,8 @@
...
+DELETE_AWS_SECRET_ACCESS_KEY_PERF = os.environ['DELETE_AWS_SECRET_ACCESS_KEY_PERF']
```
Comment (ebattat) [inline jenkins/run_policies.py:14]: DELETE_AWS_SECRET_ACCESS_KEY_PERF => AWS_SECRET_ACCESS_KEY_DELETE_PERF

**Review by ebattat — COMMENTED** [review]
Please remove delete user from github actions. we dont want to have this option on github actions

File: jenkins/run_policies.py [hunk 5]
```diff
@@ -36,6 +38,8 @@ def get_custodian_policies(type: str = None):
 
 for region in regions:
     for policy in policies:
+        if policy == 'zombie_cluster_resource':
+            os.system(f"sudo podman run --rm --name cloud-governance -e account='perf' -e policy={policy} -e AWS_ACCESS_KEY_ID={AWS_ACCESS_KEY_ID_DELETE_PERF} -e AWS_SECRET_ACCESS_KEY={AWS_SECRET_ACCESS_KEY_DELETE_PERF} -e AWS_DEFAULT_REGION={region} -e dry_run=yes -e policy_output=s3://{BUCKET_PERF}/{LOGS}/{region} -e log_level=INFO quay.io/ebattat/cloud-governance:latest")
```
Comment (ebattat) [inline jenkins/run_policies.py:42]: dry_run=no
Am I right ?

File: .github/workflows/Build.yml [hunk 6]
```diff
@@ -45,86 +46,125 @@ jobs:
...
-              current_wait_time="$((current_wait_time+sleep_time))"
-              pip --no-cache-dir install cloud-governance --upgrade
-              build=$(pip freeze | grep cloud-governance | sed 's/==/=/g')
-              actual_version="$(cut -d'=' -f2 <<<"$build")"
-              if (( $current_wait_time == $timeout )); then
-                echo "cloud-governance lastet version did not update in Pypi after 900 seconds - raise failure"
-                exit 1
+
+  build:
+
```
Comment (ebattat) [inline .github/workflows/Build.yml:51]: extra enter

File: .github/workflows/Build.yml [hunk 7]
```diff
@@ -45,86 +46,125 @@ jobs:
...
+    name: build
+    needs: [ test ]
+    runs-on: ubuntu-latest
+
```
Comment (ebattat) [inline .github/workflows/Build.yml:55]: extra enter

File: .github/workflows/Build.yml [hunk 8]
```diff
@@ -45,86 +46,125 @@ jobs:
...
+      - name: ⌛ Upload to 🐋 quay.io
+        run: |
+          version=$(python3 setup.py --version)
+          sudo podman build --build-arg VERSION=$version --layers=false -t ${{ secrets.PACKAGE_NAME }}:v$version .
+          sudo podman login quay.io -u ${{ secrets.QAUYIO_ROBOT_USER }} -p ${{ secrets.QAUYIO_ROBOT_TOKEN }}
+          sudo podman push localhost/${{ secrets.PACKAGE_NAME }}:v$version quay.io/${{ secrets.QAUYIO_REPOSITORY }}/${{ secrets.PACKAGE_NAME }}:v$version
+          sudo podman push localhost/${{ secrets.PACKAGE_NAME }}:v$version quay.io/${{ secrets.QAUYIO_REPOSITORY }}/${{ secrets.PACKAGE_NAME }}
+          echo 'Wait 30 sec till image will be updated in quay.io'
+          sleep 30
+
```
Comment (ebattat) [inline .github/workflows/Build.yml:112]: extra enter

File: .github/workflows/Build.yml [hunk 9]
```diff
@@ -45,86 +46,125 @@ jobs:
...
+  e2e:
+
```
Comment (ebattat) [inline .github/workflows/Build.yml:114]: extra enter

File: .github/workflows/Build.yml [hunk 10]
```diff
@@ -45,86 +46,125 @@ jobs:
...
+      name: e2e
+      runs-on: ubuntu-latest
+      needs: [ test, build ]
+
```
Comment (ebattat) [inline .github/workflows/Build.yml:118]: extra enter

File: .github/workflows/Build.yml (hunk 6 above)
Comment (athiruma) [reply .github/workflows/Build.yml:51]: You gave space in the benchmark-runner. That's why I add it. Space is ok, so we can see it is another job.

Comment (ebattat) [reply .github/workflows/Build.yml:51]: OK

File: .github/workflows/Build.yml [hunk 11]
```diff
@@ -45,86 +46,125 @@ jobs:
...
+             done
+          done
+          
+          # gileaks policy
+          # GITHUB_TOKEN already taken => GIT_TOKEN
+          region='us-east-1'
+          policy='gitleaks' 
+          sudo podman run --rm --name cloud-governance -e policy=$policy -e AWS_ACCESS_KEY_ID=$AWS_ACCESS_KEY_ID -e AWS_SECRET_ACCESS_KEY=$AWS_SECRET_ACCESS_KEY -e AWS_DEFAULT_REGION=$region -e git_access_token=${{ secrets.GIT_TOKEN }} -e git_repo=https://github.com/redhat-performance -e several_repos=yes -e policy_output=${{ secrets.OUTPUT_BUCKET }}/test/$region -e log_level=INFO quay.io/ebattat/cloud-governance
+
+  bump_version:
```
Comment (ebattat) [inline .github/workflows/Build.yml:143]: move bump_version job before e2e

File: jenkins/run_policies.py [hunk 12]
```diff
@@ -36,6 +38,8 @@ def get_custodian_policies(type: str = None):
 
 for region in regions:
     for policy in policies:
+        if policy == 'zombie_cluster_resource':
+            os.system(f"sudo podman run --rm --name cloud-governance -e account='perf' -e policy={policy} -e AWS_ACCESS_KEY_ID={AWS_ACCESS_KEY_ID_DELETE_PERF} -e AWS_SECRET_ACCESS_KEY={AWS_SECRET_ACCESS_KEY_DELETE_PERF} -e AWS_DEFAULT_REGION={region} -e dry_run=no -e policy_output=s3://{BUCKET_PERF}/{LOGS}/{region} -e log_level=INFO quay.io/ebattat/cloud-governance:latest")
```
Comment (ebattat) [inline jenkins/run_policies.py:42]: Dry_run =no
Also add comment before if: Delete zombie cluster resource every night

File: .github/workflows/Build.yml [hunk 13]
```diff
@@ -45,86 +46,125 @@ jobs:
...
+
+      steps:
+        - uses: actions/checkout@v2
+        - name: Set up Python 3.9
+          uses: actions/setup-python@v2
+          with:
+            python-version: 3.9
+      - name: ✔️ E2E tests using latest quay.io
+        run: |
+          declare -a regions=('us-east-1' 'us-east-2' 'us-west-1' 'us-west-2' 'eu-central-1')
```
Comment (ebattat) [inline .github/workflows/Build.yml:127]: Add matrix strategy for region and policy

Comment (athiruma) [reply .github/workflows/Build.yml:127]: ok sir.

File: jenkins/run_policies.py (hunk 12 above)
Comment (athiruma) [reply jenkins/run_policies.py:43]: ok sir.

File: jenkins/Jenkinsfile [hunk 14]
```diff
@@ -3,6 +3,8 @@ pipeline {
     environment {
         AWS_ACCESS_KEY_ID_PERF = credentials('cloud-governance-aws-access-key-id-perf')
         AWS_SECRET_ACCESS_KEY_PERF = credentials('cloud-governance-aws-secret-access-key-perf')
+        AWS_ACCESS_KEY_ID_DELETE_PERF = credentials('delete-cloud-governance-aws-access-key-id-delete-perf')
```
Comment (ebattat) [inline jenkins/Jenkinsfile:6]: delete-cloud-governance-aws-access-key-id-delete-perf = > cloud-governance-aws-access-key-id-delete-perf

File: jenkins/run_policies.py [hunk 15]
```diff
@@ -3,6 +3,8 @@
 
 AWS_ACCESS_KEY_ID_PERF = os.environ['AWS_ACCESS_KEY_ID_PERF']
 AWS_SECRET_ACCESS_KEY_PERF = os.environ['AWS_SECRET_ACCESS_KEY_PERF']
+AWS_ACCESS_KEY_ID_DELETE_PERF = os.environ['DELETE_AWS_ACCESS_KEY_ID_DELETE_PERF']
```
Comment (ebattat) [inline jenkins/run_policies.py:6]: DELETE_AWS_ACCESS_KEY_ID_DELETE_PERF => AWS_ACCESS_KEY_ID_DELETE_PERF

File: jenkins/run_policies.py [hunk 16]
```diff
@@ -3,6 +3,8 @@
...
+AWS_SECRET_ACCESS_KEY_DELETE_PERF = os.environ['DELETE_AWS_SECRET_ACCESS_KEY_DELETE_PERF']
```
Comment (ebattat) [inline jenkins/run_policies.py:7]: DELETE_AWS_SECRET_ACCESS_KEY_DELETE_PERF => AWS_SECRET_ACCESS_KEY_DELETE_PERF

File: .github/workflows/PR.yml [hunk 17]
```diff
@@ -9,6 +9,7 @@ on:
 
 jobs:
   pytest:
```
Comment (ebattat) [inline .github/workflows/PR.yml:11]: pytest => test

File: .github/workflows/PR.yml [hunk 18]
```diff
@@ -9,6 +9,7 @@ on:
...
+    name: pytest
```
Comment (ebattat) [inline .github/workflows/PR.yml:12]: pytest => test

File: .github/workflows/PR.yml [hunk 19]
```diff
@@ -45,12 +46,45 @@ jobs:
...
-        policy: ${{ secrets.POLICY }}
-        policy_output: ${{ secrets.POLICY_OUTPUT }}
-        RUNNER_PATH: ${{ secrets.RUNNER_PATH }}
-      run: |
-        PYTHONPATH=. python cloud_governance/main/main.py 1> /dev/null 2> $RUNNER_PATH/err.log
-        if [ -s "$RUNNER_PATH/err.log" ]; then echo "Error E2E raised"; fi
+
+  e2e:
+    name: e2e
+    needs: [ pytest ]
```
Comment (ebattat) [inline .github/workflows/PR.yml:52]: needs: [ test ]

File: jenkins/Jenkinsfile [hunk 20]
```diff
@@ -3,6 +3,8 @@ pipeline {
     environment {
         AWS_ACCESS_KEY_ID_PERF = credentials('cloud-governance-aws-access-key-id-perf')
         AWS_SECRET_ACCESS_KEY_PERF = credentials('cloud-governance-aws-secret-access-key-perf')
+        AWS_ACCESS_KEY_ID_DELETE_PERF = credentials('cloud-governance-aws-access-key-id-delete-perf')
```
Comment (ebattat) [inline jenkins/Jenkinsfile:6]: Did you update it in Jenkins secret ?

Comment (athiruma) [reply jenkins/Jenkinsfile:6]: yes sir. updated.

File: .github/workflows/Build.yml [hunk 21]
```diff
@@ -45,86 +46,120 @@ jobs:
...
+          git push https://${{ secrets.GITHUB_TOKEN }}@${{ secrets.GIT_REPOSITORY}} --tag
+  e2e:
+
+      name: e2e
+      runs-on: ubuntu-latest
+      needs: [ test, build, bump_version ]
+      strategy:
+        matrix:
+          region: [ 'us-east-1' 'us-east-2' 'us-west-1' 'us-west-2' 'eu-central-1' ]
+          policy: [ 'ec2_idle' 'ec2_run' 'ebs_unattached' 'ebs_in_use'  ]
```
Comment (ebattat) [inline .github/workflows/Build.yml:147]: Do we want to test it on python-version: [ '3.8', '3.9' ] ?

**Review by ebattat — COMMENTED** [review]
/LGTM
Need to verify that all tests is passed

File: .github/workflows/Build.yml [hunk 22]
```diff
@@ -40,91 +43,119 @@ jobs:
...
+
+  e2e:
+
+      name: e2e
+      runs-on: ubuntu-latest
+      needs: [ test, build, bump_version ]
+      strategy:
+        matrix:
+          region: [ 'us-east-1' 'us-east-2' 'us-west-1' 'us-west-2' 'eu-central-1' ]
+          policy: [ 'ec2_idle' 'ec2_run' 'ebs_unattached' 'ebs_in_use'  ]
```
Comment (athiruma) [inline .github/workflows/Build.yml:149]: No zombie_cluster_resource. Do I add or not.

File: .github/workflows/Build.yml [hunk 23]
```diff
@@ -40,91 +43,120 @@ jobs:
...
+          git push https://${{ secrets.GITHUB_TOKEN }}@${{ secrets.GIT_REPOSITORY}} --tag
+
+  e2e:
+
+      name: e2e
+      runs-on: ubuntu-latest
+      needs: [ test, build, bump_version ]
+      strategy:
+        matrix:
+          region: [ 'us-east-1' 'us-east-2' 'us-west-1' 'us-west-2' 'eu-central-1', 'ap-south-1', 'eu-north-1', 'ap-northeast-1', 'ap-southeast-1', 'ap-southeast-2', 'eu-south-1', 'eu-west-3', 'sa-east-1' ]
```
Comment (ebattat) [inline .github/workflows/Build.yml:148]: Remove the comma between resource !!!

**Review by ebattat — COMMENTED** [review]
Please fix it !!!

File: cloud_governance/main/main.py [hunk 24]
```diff
@@ -12,7 +12,7 @@
 from cloud_governance.common.aws.s3.s3_operations import S3Operations
 
 # env tests
-# os.environ['AWS_DEFAULT_REGION'] = 'us-east-2'
+# os.environ['AWS_DEFAULT_REGION'] = 'us-west-2'
```
Comment (ebattat) [inline cloud_governance/main/main.py:15]: # os.environ['AWS_DEFAULT_REGION'] = 'us-east-2'

File: .github/workflows/Build.yml [hunk 25]
```diff
@@ -40,91 +42,112 @@ jobs:
...
+          git config --global user.email ${{ secrets.EMAIL }}
+          git config --global user.name  ${{ secrets.NAME }}
+          git config pull.rebase false  # merge (the default strategy)
+          bumpversion patch
+          # GITHUB_REPOSITORY already taken => GIT_REPOSITORY
+          git commit .bumpversion.cfg setup.py -m 'bump version to exist version v'$version
+          git pull https://${{ secrets.GITHUB_TOKEN }}@${{ secrets.GIT_REPOSITORY}} main
+          git push https://${{ secrets.GITHUB_TOKEN }}@${{ secrets.GIT_REPOSITORY}} main
+          git push https://${{ secrets.GITHUB_TOKEN }}@${{ secrets.GIT_REPOSITORY}} --tag
+  e2e:
```
Comment (ebattat) [inline .github/workflows/Build.yml:135]: Enter

Comment (athiruma) [reply .github/workflows/Build.yml:136]: yes sir. Fixed

**Pull request merged**